compat/cpu.h

```diff
@@ -39,45 +35,4 @@ inline int sched_getaffinity(int, size_t, cpu_set_t *) { return ENOTSUP; }
 #define CPU_SET(cpu, set) (void)0
 #endif
 
-#if !defined(CPU_CLR)
-#define CPU_CLR(cpu, set) (void)0
-#endif
-
-#if !defined(CPU_ISSET)
-#define CPU_ISSET(cpu, set) false
-#endif
-
-// glibc prior to 2.6 lacks CPU_COUNT
-#ifndef CPU_COUNT
-#define CPU_COUNT(set) CpuCount(set)
```

Comment:
This PR removes a definition of `CPU_COUNT`. Our src/CpuAffinitySet.cc is using that macro and is compiled unconditionally. AFAICT, removing this macro will break the build on platforms without CPU affinity support. I and 96c2bb6 assume there are such supported platforms but please correct me if I am wrong.

FYI, according to 96c2bb6:

> Squid builds on systems without Linux CPU affinity calls, but setting affinity only works if there are sched_getaffinity(2) and sched_setaffinity(2) available. If there is no OS support but cpu_affinity options are configured, Squid quits with an error. If there is OS support but calls fail, Squid prints an error but does not quit.

I did not check other removed macros.

IMHO, this cleanup should not be done without testing on a system without Linux CPU affinity calls.

> `glibc prior to 2.6 lacks CPU_COUNT`

FWIW, this comment should not be misinterpreted as "every supported build environment has `CPU_COUNT` except glibc prior to 2.6".

Reply:
> FWIW, this comment should not be misinterpreted as "every supported build environment has CPU_COUNT except glibc prior to 2.6".

Doh, you're right.
FreeBSD has it in <sys/cpuset.h>
OpenBSD seems not to have it.

At the same time , CpuCount goes a very roundabout way; if a plaform doesn't have CPU_COUNT it's unlikely to have CPU_ISSET, so CpuCount is a very roundabout way to return 0. How about just defining to 0 when CPU_COUNT is not defined?


This patch goes a bit of a